### PR TITLE
Enables job configuration import (#6084)

### PR DESCRIPTION
**Pull request by volsch**
- Partial backport from 2.32 (just minimum required functionality for 2.29,
  users should not get issues when upgrading to latest 2.29 build).

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/DefaultMetadataExportService.java [hunk 1]
```diff
@@ -269,7 +270,7 @@ public MetadataExportParams getParamsFromMap( Map<String, List<String>> paramete
             Class<? extends IdentifiableObject> klass = (Class<? extends IdentifiableObject>) schema.getKlass();
 
             // class is enabled if value = true, or fields/filter/order is present
-            if ( "true".equalsIgnoreCase( parameters.get( parameterKey ).get( 0 ) ) || (parameter.length > 1 && ("fields".equalsIgnoreCase( parameter[1] )
+            if ( isSelectedClass( parameters.get( parameterKey ) ) || ( parameter.length > 1 && ( "fields".equalsIgnoreCase( parameter[1] )
```
Comment (huzvak) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/DefaultMetadataExportService.java]: Wouldn't it change the final logic when you now check the whole collection and not only the first item? If you are sure about it that it won't change it or if your change is made for a reason, then fine.

File: dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/metadata/DefaultMetadataExportServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    @InjectMocks
+    private DefaultMetadataExportService service;
+
+    @Rule
+    public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+    @Test
+    public void getParamsFromMapIncludedSecondary()
+    {
+        Mockito.when( schemaService.getSchemaByPluralName( Mockito.eq( "jobConfigurations" ) ) )
```
Comment (huzvak) [inline dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/metadata/DefaultMetadataExportServiceTest.java]: Those two `Mockito.when` statements are present in all 3 tests. You could move it into before method. But it is up to you. Your tests class so it is up to you what you find better/more readable/nicer. Logic is OK.

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/DefaultMetadataExportService.java (hunk 1 above)
Comment (volsch) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/DefaultMetadataExportService.java]: The front-end app for 2.30 and later has a work around for job configurations in order not to show them in the export front-end. This causes that jobConfigurations=true&jobConfigurations=false (which makes no sense, there should be just one of them) is added.

Comment (huzvak) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/DefaultMetadataExportService.java]: Cool. If you have justification for it then fine. Makes sense.

**Pull request merged**